### PR TITLE
Adding recovery phrase video to onboarding process

### PR DESCRIPTION
**Pull request by ryanml**
Related: https://github.com/MetaMask/metamask-extension/issues/10244

Screen Shot 2021-03-25 at 2 47 23 AM

Manual testing steps:  
  - Clean install MetaMask
  - Select the "Create Account Route"
  - Confirm after creating a password you are directed to the seed phrase video page
  - Confirm that clicking next further continues on to the reveal seed phrase page
  - Confirm that this component is not shown in the "Import Account flow" on fresh install"

### CODE REVIEW

**Comment by kumavis** [comment]
:sunglasses: :sunglasses: :sunglasses:

**Comment by kumavis** [comment]
note: video is delivered as part of extension, adds ~3mb (seems reasonable)

**Comment by kumavis** [comment]
how to intlize video? is it possible to include and specify alternate audio tracks?

**Comment by ryanml** [comment]
> how to intlize video? is it possible to include and specify alternate audio tracks?

@kumavis see https://consensys.slack.com/archives/GETEW1V4M/p1616604943116400 for relevant discussion, I think we'll follow up on this

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase.component.js [hunk 1]
```diff
@@ -39,12 +41,15 @@ export default class SeedPhrase extends PureComponent {
...
+    const { seedPhrase, history } = this.props;
     const { verifiedSeedPhrase } = this.state;
+    const pathname = history?.location?.pathname;
+    const introClass =
+      pathname === INITIALIZE_SEED_PHRASE_INTRO_ROUTE ? 'intro' : '';
 
     return (
       <DragDropContextProvider backend={HTML5Backend}>
-        <div className="first-time-flow__wrapper">
+        <div className={`first-time-flow__wrapper ${introClass}`}>
```
Comment (ryanml) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase.component.js:52]: When we modify the rest of the onboarding flow to follow the new designs, we can just bake rules directly in to `first-time-flow__wrapper`. Presently, it doesn't play so nicely with the new video screen

**Review by rachelcope — DISMISSED** [review]
Looks good to me!

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+// Components
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../../../components/ui/button';
+// Routes
+import { INITIALIZE_SEED_PHRASE_ROUTE } from '../../../../helpers/constants/routes';
+
+export default class SeedPhraseIntro extends PureComponent {
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js:8]: [REQ] We don't want to extend the pattern of using class components and splitting into containers/components. Could you refactor this into a functional component?

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      </>
+    );
+  }
+
+  renderSidebarContent() {
+    const { t } = this.context;
+
+    return (
+      <div className="seed-phrase-intro__sidebar">
+        <div className="seed-phrase-intro__sidebar-block">
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js:61]: Try to use `<Box>` for some of these divs. Box is a layout component with various props for controlling size/spacing/layout.

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+          <div className="seed-phrase-intro__sidebar-title">
+            {t('seedPhraseIntroSidebarTitleOne')}
+          </div>
+          <div className="seed-phrase-intro__sidebar_copy">
+            {t('seedPhraseIntroSidebarCopyOne')}
+          </div>
+        </div>
+        <div className="seed-phrase-intro__sidebar-block">
+          <div className="seed-phrase-intro__sidebar-title">
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js:70]: Try using `<Typography>` and semantic html tags like heading tags/p tags/span tags where appropriate.

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  &__next-wrapper {
+    width: 325px;
+  }
+
+  &__sidebar {
+    padding: 15px;
+    border-radius: 8px;
+    border: 1px solid #d6d9dc;
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss:59]: Don't use hex codes, match them to the color from the design system colors file. In this case $ui-2. https://github.com/MetaMask/metamask-extension/blob/12c614cc14fa9a146170e1989726deb758da310b/ui/app/css/design-system/colors.scss#L78-L110

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  &__right {
+    flex: 1;
+    min-width: 0;
+
+    @media screen and (max-width: 830px) {
+      margin-top: 24px;
+    }
+  }
+
+  &__title {
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss:28]: This class, for example, can be expressed with Typography
```jsx
<Typography color={COLORS.BLACK} variant={TYPOGRAPHY.H1} boxProps={{ marginTop: 0, marginBottom: 4 }} >
```

**Review by brad-decker — APPROVED** [review]
Looks awesome, a few change reqs but overall very excited about this landing.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Oops, was meant to hit req changes ;P

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js (hunk 2 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js:8]: Force of habit ;) Can do!

**Review by Az111z — APPROVED** [review]

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js (hunk 2 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js]: done

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js (hunk 3 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js]: Using box where it seems to make sense now

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js (hunk 4 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js]: done!

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss (hunk 6 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss]: Done, and this rule removed as a result

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss (hunk 5 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/index.scss]: done

**Comment by ryanml** [comment]
Changes pushed cc: @brad-decker - resulting in a 77 LOC reduction in the diff

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+              </span>
+              <ul className="seed-phrase-intro__sidebar_list">
+                <li>{t('seedPhraseIntroSidebarBulletOne')}</li>
+                <li>{t('seedPhraseIntroSidebarBulletTwo')}</li>
+                <li>{t('seedPhraseIntroSidebarBulletThree')}</li>
+                <li>{t('seedPhraseIntroSidebarBulletFour')}</li>
+              </ul>
+            </Box>
+            <Box marginBottom={4}>
+              <span className="seed-phrase-intro__sidebar-title">
```
Comment (brad-decker) [inline ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js:84]: Question: Why could the sidebar-title class items not be converted into Typography components?

**Review by brad-decker — DISMISSED** [review]
LGTM, just a question

File: ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js (hunk 7 above)
Comment (ryanml) [reply ui/app/pages/first-time-flow/seed-phrase/seed-phrase-intro/seed-phrase-intro.component.js]: `seed-phrase-intro__sidebar-title` css removed and elements converted to use the `Typography` component 👍

**Review by brad-decker — APPROVED** [review]
LGTM!

**Pull request merged**